File: src/main/java/net/gegy1000/terrarium/server/map/source/LoadingState.java
package net.gegy1000.terrarium.server.map.source;

public enum LoadingState {
    LOADING_CACHED("state.cached", 0, 1),
    LOADING_ONLINE("state.online", 10, 3),
    LOADING_NO_CONNECTION("state.no_connection", 20, 5);

    private final String languageKey;
    private final int textureY;
    private final int weight;

    LoadingState(String languageKey, int textureY, int weight) {
        this.languageKey = languageKey;
        this.textureY = textureY;
        this.weight = weight;
    }

    public String getLanguageKey() {
        return this.languageKey;
    }

    public int getTextureY() {
        return this.textureY;
    }

    public int getWeight() {
        return this.weight;
    }
}

File: src/main/kotlin/net/gegy1000/terrarium/server/map/source/LoadingState.kt
package net.gegy1000.terrarium.server.map.source

enum class LoadingState(val languageKey: String, val textureY: Int) {
    LOADING_CACHED("state.cached", 0),
    LOADING_ONLINE("state.online", 10),
    LOADING_NO_CONNECTION("state.no_connection", 20)
}
